src/Razor/src/Microsoft.VisualStudio.Editor.Razor/RazorTextViewConnectionListener.cs

```diff
@@ -48,8 +44,10 @@ public async void SubjectBuffersConnected(ITextView textView, ConnectionReason r
                     throw new ArgumentNullException(nameof(subjectBuffers));
                 }
 
-                _joinableTaskContext.AssertUIThread();
-                await _documentManager.OnTextViewOpenedAsync(textView, subjectBuffers);
+                _joinableTaskContext.Factory.Run(async () => {
+                    _joinableTaskContext.AssertUIThread();
```

Comment:
Does this work? I would have thought you'd want to switch to the main thread, not assert, since you're in an async lambda. Switching will just no-op if you're already there.

Reply:
That's a good point, @NTaylorMullen do you know how I activate this codepath to test it?

Reply:
This is old editor only. You'd need to enable the legacy editor and then it should trigger upon opening a Razor file

Reply:
I ran through this a couple times in the legacy editor and it does in fact work without throwing. I do see your point though, @NTaylorMullen is this dangerous and should be done a different way?

Reply:
This API is expected to be called on the UI thread so I wouldn't say it's dangerous. Makes it a bit more clear on our expectations as to what thread this should operate on. Now that being said I would put the "AssertUIThread" bit outside of the `.Run`